src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -376,4 +430,38 @@ public String getWriteableName() {
     private static boolean isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport() {
         return NeuralSearchClusterUtil.instance().getClusterMinVersion().onOrAfter(MINIMAL_SUPPORTED_VERSION_DEFAULT_MODEL_ID);
     }
+
+    private Map<String, Float> getAllTokens() {
+        Map<String, Float> queryTokens = queryTokensSupplier.get();
+        validateQueryTokens(queryTokens);
+        return queryTokens;
+    }
+
+    private Map<String, Float> getHighScoreTokens(Map<String, Float> queryTokens, float ratio) {
+        return getFilteredScoreTokens(queryTokens, true, ratio);
+    }
+
+    private Map<String, Float> getLowScoreTokens(Map<String, Float> queryTokens, float ratio) {
+        return getFilteredScoreTokens(queryTokens, false, ratio);
+    }
+
+    private Map<String, Float> getFilteredScoreTokens(Map<String, Float> queryTokens, boolean aboveThreshold, float ratio) {
+        float max = queryTokens.values().stream().max(Float::compare).orElse(0f);
+        float threshold = ratio * max;
+        if (max == 0) {
+            return Collections.emptyMap();
+        }
+        return queryTokens.entrySet()
+            .stream()
+            .filter(entry -> (aboveThreshold == (entry.getValue() >= threshold)))
```

Comment:
please add comment for logic behind this call

Reply:
nit: you should put comment before the code, not a blocker but please in mind for next PRs